spec/tbd_tests_spec.rb

```diff
@@ -1403,77 +1403,88 @@
-      warehouse.setName("Warehouse - fine"     )
-      offices.setName("Office - enclosed"      )
-      sales.setName("Sales area"               )
-      workshop.setName("Workshop space"        )
-
-      expect(north.setSpaceType(audience )).to be(true)
-      expect( east.setSpaceType(warehouse)).to be(true)
-      expect(south.setSpaceType(offices  )).to be(true)
-      expect( west.setSpaceType(sales    )).to be(true)
-      expect( core.setSpaceType(workshop )).to be(true)
+    # Assign (missing) space types.
+    north = model.getSpaceByName("Story 1 North Perimeter Space")
+    east  = model.getSpaceByName("Story 1 East Perimeter Space" )
+    south = model.getSpaceByName("Story 1 South Perimeter Space")
+    west  = model.getSpaceByName("Story 1 West Perimeter Space" )
+    core  = model.getSpaceByName("Story 1 Core Space"           )
+
+    expect(north.empty?).to be(false)
+    expect(east.empty? ).to be(false)
+    expect(south.empty?).to be(false)
+    expect(west.empty? ).to be(false)
+    expect(core.empty? ).to be(false)
+
+    north = north.get
+    east  = east.get
+    south = south.get
+    west  = west.get
+    core  = core.get
+
+    audience  = OpenStudio::Model::SpaceType.new(model)
+    warehouse = OpenStudio::Model::SpaceType.new(model)
+    offices   = OpenStudio::Model::SpaceType.new(model)
+    sales     = OpenStudio::Model::SpaceType.new(model)
+    workshop  = OpenStudio::Model::SpaceType.new(model)
+
+    audience.setName("Audience - auditorium" )
+    warehouse.setName("Warehouse - fine"     )
+    offices.setName("Office - enclosed"      )
+    sales.setName("Sales area"               )
+    workshop.setName("Workshop space"        )
+
+    expect(north.setSpaceType(audience )).to be(true)
+    expect( east.setSpaceType(warehouse)).to be(true)
+    expect(south.setSpaceType(offices  )).to be(true)
+    expect( west.setSpaceType(sales    )).to be(true)
+    expect( core.setSpaceType(workshop )).to be(true)
 
-      json = TBD.process(model, argh)
-      expect(json.is_a?(Hash)    ).to be(true)
-      expect(json.key?(:io      )).to be(true)
-      expect(json.key?(:surfaces)).to be(true)
+    json = TBD.process(model, argh)
+    expect(json.is_a?(Hash)    ).to be(true)
+    expect(json.key?(:io      )).to be(true)
+    expect(json.key?(:surfaces)).to be(true)
 
-      io        = json[:io      ]
-      surfaces  = json[:surfaces]
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(argh.key?(:roof_uo)).to be(false)
+
+    # OpenStudio prior to v3.5.X had a 3m maximum layer thickness, reflecting a
+    # previous v8.8 EnergyPlus constraint. TBD caught such cases when uprating
+    # (as per NECB requirements). From v3.5.0+, OpenStudio dropped the maximum
+    # layer thickness limit, harmonizing with EnergyPlus:
+    #
+    #   https://github.com/NREL/OpenStudio/pull/4622
+    if OpenStudio.openStudioVersion.split(".").join.to_i < 350
       expect(TBD.error?     ).to be( true)
```

Comment:
TBD only raises max layer thickness errors for OpenStudio versions prior to v3.5.1.